core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -482,6 +499,34 @@ object UnitActions {
         )
     }
 
+    private fun addRepairAction(unit: MapUnit, actionList: ArrayList<UnitAction>) {
+        if (!unit.hasUniqueToBuildImprovements) return
+        if (unit.isEmbarked()) return
+        val tile = unit.getTile()
+        if (!(tile.improvementIsPillaged || tile.roadIsPillaged)) return
+
+        val couldConstruct = unit.currentMovement > 0
+                && !tile.isCityCenter()
+
+        actionList += UnitAction(UnitActionType.Repair,
+            action = getRepairAction(unit).takeIf { couldConstruct }
+        )
+    }
+
+    fun getRepairAction(unit: MapUnit): () -> Unit {
+        return {
+            unit.action = null
+            unit.currentMovement = 0f
+            val tile = unit.currentTile
+            if (tile.improvementIsPillaged)
+                tile.improvementIsPillaged = false
+            else
+                tile.roadIsPillaged = false
```

Comment:
Shouldn't repairing take 3 turns instead of being instant?

Reply:
I had tried on Civ 5 G&K and it was instant

Reply:
What game speed did you try it on? If it was 'quick', I wouldn't be surprised if it instantly repaired it, but maybe on marathon it doesn't

Reply:
Just checked again. Regular speed is 2 turns

Reply:
K, now 2 turn Repair. Need to figure out the values on all speeds

Reply:
I'd gamble that its 3 * gameSpeed.modifier, as almost everything is that

Reply:
Well the code comment claims 3, but Standard Speed G&K is 2. I see the scaling factors in the code but hoping there's some XML values somewhere

Reply:
Nope, tried Vanilla Standard, also 2 turns. Weird